src/routes/customers.php

```diff
@@ -15,8 +15,147 @@
         $stmt = $db->query($sql);
         $customers = $stmt->fetchAll(PDO::FETCH_OBJ);
         $db = null;
-        echo json_encode($customers);
+        $response->write(json_encode($customers));
+        return $response->withStatus(200);
     } catch (PDOException $e) {
-        echo '{"error": {"text": ' . $e->getMessage() . '}';
+        $db = null;
+        $response->write($e);
+        return $response->withStatus(500);
+    }
+});
+
+// Get Customer By Username
+$app->get('/api/customers/{username}', function (Request $request, Response $response, array $args) {
+    $username = $args['username'];
+    $sql = "SELECT * FROM customers WHERE username = :username";
+
+    try {
+        // Get DB Object
+        $db = new db();
+
+        $db = $db->connect();
+        $stmt = $db->prepare($sql);
+        $stmt->bindParam(':username', $username);
+        $stmt->execute();
+        $customer = $stmt->fetch(PDO::FETCH_OBJ);
```

Comment:
you need to add `if(customer)` here to determine if the object exist or not, if not return 404 with an error message,
 because this way even if the requested resource doesn't exist, it will still return 200 with nothing it it.